backend/src/main/java/com/exadel/carinsurance/service/implementation/AuthService.java

```diff
@@ -85,16 +88,28 @@ public ResponseEntity login( @RequestBody AuthRequestEntity request ) {
         )
     );
 
-    UserEntity user = ( UserEntity ) authentication.getPrincipal();
+    UserEntity userEntity = ( UserEntity ) authentication.getPrincipal();
+    UserResponseEntity userResponse = UserResponseEntity.fromUserEntity( userEntity );
     String jwtToken = jwtService.generateToken( userEmail );
 
+    ResponseCookie cookie = ResponseCookie.from( "token", jwtToken )
+        .path( "/" )
+        .maxAge( 86400 )
+        .build();
+
+    HttpHeaders headers = new HttpHeaders();
+    headers.add( HttpHeaders.SET_COOKIE, cookie.toString() );
+    headers.add( HttpHeaders.ACCESS_CONTROL_EXPOSE_HEADERS, "Set-Cookie" );

```

Comment:
[Please ignore] No idea why you want to expose cookies as a part of CORS. This does looks like a major vurability for CSRF attack. But as this is outside of our initial learning scope you can ignore my comment